src/Machete/Values/EntityValueList.cs

```diff
@@ -51,6 +51,9 @@ public bool TryGetValue(int index, out Value<TValue> value)
                 index >= _values.Count && _slice.TryGetSlice(i, out textSlice);
                 i++)
             {
+                if (textSlice.SourceSpan.IsEmpty)
```

Comment:
I don't think this will do what is needed, since there can be empty slices in repeating components.